trackintel/analysis/transport_mode_identification.py

```diff
@@ -0,0 +1,62 @@
+
+import trackintel as ti
+from trackintel.geogr.distances import haversine_dist
+import geopandas as gpd
+import warnings
+
+
+def predict_transport_mode(triplegs, method='simple-coarse'):
+    """
+    Predicts/imputes the transport mode that was likely chosen to cover the 
+    given tripleg, e.g., car, bicycle, or walk.
+
+    Parameters
+    ----------
+    method: str, {'simple-coarse'}, default 'simple-coarse'
+        The following methods are available for transport mode inference/prediction:
+        'simple-coarse' : Uses simple heuristics to predict coarse transport classes.
+            These include ``{'slow_mobility', 'motorized_mobility', 'fast_mobility'}``.
+            In this classification, ``slow_mobility`` includes transport modes such as
+            walking or cycling, ``motorized_mobility`` modes such as car or train, and
+            ``fast_mobility`` modes such as high-speed rail or airplanes.
+    """
+    if method == 'simple-coarse':
+        return predict_transport_mode_simple_coarse(triplegs)
+    else:
+        raise NameError(f'Method {method} not known for predicting tripleg transport modes.')
+
+
+def predict_transport_mode_simple_coarse(triplegs):
+    """
+    Predicts a transport mode out of three coarse classes. Implements a simple speed-
+    based heuristic (over the whole tripleg) that uses the cutoffs 15 km/h and 100 km/h. 
+    As such, it is very fast, but also very simple and coarse.
+
+    For additional documentation, see 
+    :func:`trackintel.analysis.transport_mode_identification.predict_transport_mode`.
+    """
+    triplegs = triplegs.copy()
+    
+    wgs = ti.geogr.distances.check_crs(triplegs)
+
+
+    def identify_mode(tripleg,wgs):
```

Comment:
This looks like an indentation error or is this because of the online viewer?

Reply:
I think Dominik implemented the function inside the other function. Idk if this is correct, but it seems to be working. Indent error would have caused the tests to fail, wouldn't it?

Reply:
Yes, I think [nested functions](https://realpython.com/inner-functions-what-are-they-good-for/) are OK. They are not mentioned in pep8 so why not